### ISSUE TITLE
Directory and files within are missing after rename of folder

### ISSUE THREAD

**Issue by shaneonabike** [owncloud/core#16925]
### Steps to reproduce

1. Create a new directory online
2. Add files into the directory
3. Share the directory to another user with all permissions
4. The second user renames the folder
5. Suddenly the folder is not available to either user

### Expected behaviour
The folder would be renamed if the correct permission exists and everything works properly

### Actual behaviour
In this case, it seems that the folder was renamed on the server but not synced. I looked on Unix and noticed the folder was there but for some reason no longer available on the web interface nor through the client software.

### Server configuration

**Web server:**
Unix/Debian

**Database:**
Mysql

**PHP version:**

**ownCloud version:** (see ownCloud admin page)
Latest

**Updated from an older ownCloud or fresh install:**

**List of activated apps:**
* Contacts
* Calendar
* Files
* Documents

**The content of config/config.php:**
```
<?php
$CONFIG = array (
  'trusted_domains' =>
  array (
    0 => 'cloud.blah.com',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'http://cloud.blah.com',
  'dbtype' => 'mysql',
  'version' => '8.0.4.2',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => 'xxxx!',
  'installed' => true,
  'mail_from_address' => 'info',
  'mail_smtpmode' => 'smtp',
  'mail_domain' => 'blah.com',
  'mail_smtpauthtype' => 'PLAIN',
 'mail_smtphost' => 'smtp.blah.com',
  'mail_smtpport' => '587',
  'mail_smtpauth' => 1,
  'mail_smtpname' => 'owncloud@blah.com',
  'mail_smtppassword' => 'xxx',
  'mail_smtpsecure' => 'tls',
  'asset-pipeline.enabled' => true,
  'forcessl' => true,
  'theme' => '',
  'maintenance' => false,
);
```

* No encryption
* Chromium Browser

**Comment by PVince81** [comment]
Please specify your ownCloud version.
"Latest" doesn't tell us whether you are using git master (8.1 beta) or 8.0.4.

**Comment by PVince81** [comment]
Did the folder contain a lot of entries ? Did it take a long time to rename, like a minute or so ?

**Comment by shaneonabike** [comment]
Sorry I know my laptop was dying and I wanted to submit this before it died. It was unclear.

Version: 8.04

It had ony three files within it exactly!

**Comment by PVince81** [comment]
Three files, hmmm... so this is not a known issue.

**Comment by shaneonabike** [comment]
It seems almost as the the directory was not locked properly and then the syncing went all bonkers. I did notice afterwards that the files were finally on the server, but ended up missing from the owncloud of all the users as well as the Ubuntu / Mac tools that connect to owncloud.

**Comment by PVince81** [comment]
Yeah, likely a concurrency issue. Even when users don't do anything, their sync clients might be pinging the server to get a status.

This is solved now in 8.2RC1 with transactional file locking.

It can also be enabled manually (opt-in in 8.1, default enabled in 8.2), see https://doc.owncloud.org/server/8.1/admin_manual/configuration_files/files_locking_transactional.html

**Comment by PVince81** [comment]
Ref https://github.com/owncloud/core/issues/13391
